### ISSUE TITLE
Followup DB queryable format

### ISSUE THREAD

**Issue by pbiggar** [darklang/dark#4799]
These issues remain after #4797

- [ ] DvalReprInternalQueryable support for more types
  - [ ] Option/Result
  - [ ] Tuples
  - [ ] Bytes
  - [ ] tests in roundtrip
  - [ ] 
- [ ] Test Float::greaterThan with NaN/Infinity/-Infinity
- [ ] Test a Result fn with two args, one with is Ok and the other Error
- [ ] same with Option

**Comment by StachuDotNet** [comment]
folded into #5236
